File: Problems/2264. Largest 3-Same-Digit Number in String/2264. Largest 3-Same-Digit Number in String.cpp
class Solution {
public:
    string largestGoodInteger(string num) {
        char res = -1;
        char prev = -1;
        short counter = 1;

        for(char digit: num){
            if(digit == prev){
                counter += 1;
                if(counter == 3){
                    res = std::max(digit, res);
                }
            }
            else{
                prev = digit;
                counter = 1;
            }
        }

        if(res < 0) return "";
        return string(3, res);
    }
};

File: Problems/2264. Largest 3-Same-Digit Number in String/2264. Largest 3-Same-Digit Number in String.py
class Solution:
    def largestGoodInteger(self, num: str) -> str:
        res = -1;
        prev = None;
        counter = 0;
        
        for digit in num:
            if digit == prev:
                counter += 1;
                if counter == 3:
                    res = max(int(digit), res);
            else:
                prev = digit;
                counter = 1;
                
        return (
            "" if res < 0 else str(res)*3
        );